pkg/core/consensus/maintainer/maintainer.go

```diff
@@ -141,20 +142,32 @@ func (m *StakeAutomaton) sendBid() error {
 }
 
 func (m *StakeAutomaton) sendStake() error {
-	if m.amount == 0 {
-		return errors.New("zero amount")
-	}
-
-	if m.lockTime == 0 {
-		return errors.New("zero lockTime")
+	amount, lockTime := m.getTxSettings()
+	if amount == 0 || lockTime == 0 {
+		return fmt.Errorf("invalid settings: amount: %v / locktime: %v", amount, lockTime)
 	}
 
-	l.Tracef("Sending stake tx (%d,%d)", m.amount, m.lockTime)
+	l.WithFields(log.Fields{
+		"amount":   amount,
+		"locktime": lockTime,
+	}).Tracef("Sending stake tx")
 	buf := new(bytes.Buffer)
-	if err := rpcbus.MarshalConsensusTxRequest(buf, m.amount, m.lockTime); err != nil {
+	if err := rpcbus.MarshalConsensusTxRequest(buf, amount, lockTime); err != nil {
 		return err
 	}
 
 	_, err := m.rpcBus.Call(rpcbus.SendStakeTx, rpcbus.NewRequest(*buf), 0)
 	return err
 }
+
+func (m *StakeAutomaton) getTxSettings() (uint64, uint64) {
+	settings := config.Get().Consensus
```

Comment:
Assuming `settings` does not change at runtime, I would prefer to eliminate all references to `config` and inject it in the `StakeAutomaton` constructor by the client of this code

Reply:
I see. The idea was to retain some flexibility for the user, being able to adjust his values on the fly (as one's amount of DUSK can grow over time). If it is not desirable though, I can change it back to just take these values in the constructor.